### ISSUE TITLE
[Bug]: Subscribe pattern broken by default

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#62562]
### Quick summary

Subscribe pattern is broken in selector and when inserted on page

![Uploading image.png…]()

<img width="734" alt="image" src="https://user-images.githubusercontent.com/87168/162000461-5c86bea6-f9a6-4028-a106-5b90f3e3a417.png">

### Steps to reproduce

1. Open pattern sidebar
2. Explore
3. Subscribe category
4. Note one of the patterns
5. Insert on page

https://user-images.githubusercontent.com/87168/162000496-5f42dda8-074d-4f9e-a601-173f06d8a214.mov

- Test site was `fsetest296183426 dot wordpress dotcom`
- FSE theme
- Editing homepage

Lemme know if I can add you to the site as an user.

### What you expected to happen

No error.

### What actually happened

Error, recoverable with button.

### Browser

Mozilla Firefox

### Simple/Atomic

Simple

### Reproducibility

Consistent

### Available workarounds?

Yes, easy to implement

**Comment by ivan-ottinger** [comment]
There's a chance this issue is related to the recent PR: https://github.com/Automattic/jetpack/pull/23749 where the default Subscribe block was switched from _Compact_ to _Split_. As a result, the validation broke - as the older block version in patterns expects to include `"className":"is-style-compact"`.

I will check if this hypothesis is true with a couple of tests.

**Comment by simison** [comment]
Gotcha, yeah you'll need a block transform if the default output has changed.

Short term you can just update the pattern at source site to fix subscribe blocks there.

**Comment by ivan-ottinger** [comment]
For completeness, adding findings that we discussed in p1649319850929269/1649256055.377839-slack-C02TCEHP3HA.

---

Regards the cause of the issue - at the end it looks like it could have been related to [the patch](https://github.com/Automattic/jetpack/pull/21273) that introduced the “Success message”.

Missing “Success message” is the reason why the block validation fails. As soon as I add the “Success message” to the block editor, the validation passes.

The “Attempt Block Recover” tool/button does the same thing - it adds the missing “Success message”.

After further investigation, making the `is-style-compact` block style default (that I initially suspected as a culprit) did not have effect on the block validation. The class is being added to the block only after we click on one of the styles in the block settings:
![Markup on 2022-04-07 at 10:31:45](https://user-images.githubusercontent.com/25105483/162156464-dd0b6b54-c0b8-4cb0-92ab-dd12fd9e40ee.png)

And in the case of the https://github.com/Automattic/jetpack/pull/23749 PR we only made the "Split" style the default one highlighted in the editor - the block output was already using the "Split" style before the PR was merged.

---

We have now fixed the broken Subscribe pattern and it should be live soon.

**Comment by yansern** [comment]
Looking at this PR https://github.com/Automattic/jetpack/pull/21273/, this is happening because, [on the line](https://github.com/amustaque97/jetpack/blob/5710093f56c756dbfd15ca373bf71737984f876e/projects/plugins/jetpack/extensions/blocks/subscriptions/save.js#L135) that creates the `successMessage` shortcode attribute, it did not fallback to default success message coming from `defaultAttributes`.

Blocks created with older version of Subscription block do not have `successMessage` value saved in their block attributes.

Instead of:
```
success_message: successMessage,
```

Do this:
```
success_message: successMessage || defaultAttributes.successMessage.default
```

A PR should be fairly quick to make. Creating a deprecation entry is likely not necessary.

**Comment by ivan-ottinger** [comment]
Thanks for finding the root cause Yan!

> A PR should be fairly quick to make. Creating a deprecation entry is likely not necessary.

👍🏼 I will create the PR.

**Comment by ivan-ottinger** [comment]
> A PR should be fairly quick to make. Creating a deprecation entry is likely not necessary.

Unfortunately, using `success_message: successMessage || defaultAttributes.successMessage.default` on its own won't help because it is processed as a part of the block's `save()` function.

In other words, even though we have the correct `successMessage` available in `defaultAttributes.successMessage.default`, it is only passed to its shortcode attribute when we, for instance, add a new Subscribe block to the page. The change wouldn't affect blocks that were saved before.

I checked if we have any reports from users mentioning this specific issue with the Subscribe block and haven't found any. Taking this into consideration, plus that the Subscribe pattern is now fixed. Even if there are cases where the issue manifests, it can still be resolved with that one click.

I would suggest to close this issue and focus on more pressing ones. What are your thoughts? I am also happy to continue looking for a fix.
